hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -1074,6 +1046,42 @@ private TcpClientConnection onAuthenticated(TcpClientConnection connection,
         return connection;
     }
 
+    private void connectToTpcPorts(TcpClientConnection connection, List<Integer> tpcPorts) throws IOException {
+        Channel[] tpcChannels = new Channel[tpcPorts.size()];
+        for (int k = 0; k < tpcChannels.length; k++) {
+            Address address = new Address(connection.getRemoteAddress().getHost(), tpcPorts.get(k));
+            logger.info("TPC Client: Connecting to:" + address);
+            SocketChannel tpcSocketChannel = SocketChannel.open();
+
+            Channel tpcChannel = networking.register(connection.getChannelInitializer(), tpcSocketChannel, true);
+            tpcChannel.attributeMap().put(Address.class, address);
+            tpcChannel.attributeMap().put(TcpClientConnection.class, connection);
+
+            InetSocketAddress tpcSocketChannelAddress = new InetSocketAddress(address.getHost(), address.getPort());
+
+            tpcChannel.connect(tpcSocketChannelAddress, connectionTimeoutMillis);
+
+            logger.info("TPC Client: Successfully connected to " + tpcSocketChannelAddress);
+
+            tpcSocketChannel.configureBlocking(false);
```

Comment:
```suggestion
```
This seems to be unneeded due to the next line.

Reply:
Why exactly?